### DISCUSSION TITLE
How to use `rsvg_handle_render` from lua

### DISCUSSION THREAD

**Discussion by elusian** [brndnmtthws/conky#1262] [Q&A] [answered]
In Conky v1.13 the functions `rsvg_handle_render_cairo(_sub)` were removed and replaced with `rsvg_handle_render_document/layer`, which now take a `RsvgRectangle` argument for the viewport.
How do I actually create that struct from lua?

Passing a table with the correct arguments does not work ("assertion '!viewport.is_null()' failed") and the "constructors" that tolua++ should (as far as I understand) create are not available (`RsvgRectangle()` does not work because it is not callable, `RsvgRectangle:new` is nil).

I'm probably missing something very obvious (I don't have much experience with tolua++), but any help is appreciated.

**Comment by brndnmtthws** [comment, marked as answer]
This should resolve it: https://github.com/brndnmtthws/conky/pull/1263

When I updated this, I forgot to add the necessary methods.

Now you'll be able to create/destroy:

```lua
rect = RsvgRectangle::create() -- creates rectangle
rect:destroy();                -- destroys rectangle


-- alternatively, let Lua manage the memory:
rect = RsvgRectangle::create()
tolua.takeownership(rect)
-- now Lua will free the object when it goes out of scope
```

It's my bad for forgetting to add those methods.

**Reply by elusian** [reply]
Thank you!
